files/lib/data/conversation/ConversationAction.class.php

```diff
@@ -216,6 +216,11 @@ public function markAsRead() {
 		if (empty($this->parameters['visitTime'])) {
 			$this->parameters['visitTime'] = TIME_NOW;
 		}
+
+		// userID cannot be set during an AJAX request, but only when calling this method within PHP
```

Comment:
False, the user can provide arbitrary parameters.

You can however force the user id to always equal the current user's id by forcefully setting the value in the `validateMarkAsRead()` method, or throw an exception when the value is present but expected to be missing.